src/context/LogContext/index.tsx

```diff
@@ -31,42 +53,128 @@ const LogContextProvider: React.FC<LogContextProviderProps> = ({
   children,
   initialLogLines,
 }) => {
+  const [filters] = useQueryParam<string[]>(QueryParams.Filters, []);
+  const [bookmarks] = useQueryParam<number[]>(QueryParams.Bookmarks, []);
+  const [selectedLine] = useQueryParam<number | undefined>(
+    QueryParams.SelectedLine,
+    undefined
+  );
+  const [filterLogic] = useQueryParam(QueryParams.FilterLogic, FilterLogic.And);
+  const [upperRange] = useQueryParam<undefined | number>(
+    QueryParams.UpperRange,
+    undefined
+  );
+  const [lowerRange] = useQueryParam(QueryParams.LowerRange, 0);
   const { state, dispatch } = useLogState(initialLogLines);
+  const listRef = useRef<List>(null);
 
-  const ingestLines = useCallback(
-    (lines: string[], logType: LogTypes) => {
-      dispatch({ type: "INGEST_LOGS", logs: lines, logType });
-    },
-    [dispatch]
+  const getLine = useCallback(
+    (lineNumber: number) => state.logs[lineNumber],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state.logs.length]
   );
 
-  const clearLogs = useCallback(() => {
-    dispatch({ type: "CLEAR_LOGS" });
-  }, [dispatch]);
+  const scrollToLine = useCallback((lineNumber: number) => {
+    listRef.current?.scrollToRow(lineNumber);
+  }, []);
 
-  const getLine = useCallback(
-    (lineNumber: number) => state.logs[lineNumber],
-    [state.logs]
+  // TODO EVG-17537: more advanced filtering
+  const processedLogLines = useMemo(
+    () => filterLogs(state.logs, filters, bookmarks, selectedLine, filterLogic),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [state.logs.length, `${filters}`, `${bookmarks}`, selectedLine, filterLogic]
   );
 
-  const setFileName = useCallback(
-    (fileName: string) => {
-      dispatch({ type: "SET_FILE_NAME", fileName });
-    },
-    [dispatch]
+  const searchResults = useMemo(
+    () =>
+      // search through processedLoglines
+      // return the line number of the first match
+      // if no match, return undefined
+      state.searchState.searchTerm
+        ? searchLogs({
+            searchRegex: state.searchState.searchTerm,
+            processedLogLines,
+            upperBound: upperRange,
+            lowerBound: lowerRange,
```

Comment:
When it comes to the bounds, legacy Lobster has a slightly different behavior.

this PR (even if term is out of bounds, highlights it):
<img width="1259" alt="Screen Shot 2022-10-06 at 9 43 49 AM" src="https://user-images.githubusercontent.com/47064971/194329061-3a948b46-ea74-4c73-aea5-112c5053c331.png">

legacy Lobster (if term is out of bounds, there is no highlight):
<img width="1031" alt="Screen Shot 2022-10-06 at 9 44 09 AM" src="https://user-images.githubusercontent.com/47064971/194329188-28584205-3d59-4009-a3d1-951a72a19c66.png">

Reply:
Great catch!